backend/src/auth.ts

```diff
@@ -78,6 +79,21 @@ export async function authPlugin(fastify: FastifyInstance, opts: AuthPluginOpts,
     return id;
   });
 
+  fastify.addHook('preValidation', async (req) => {
+    if (opts.publicRouterPaths.has(req.routerPath)) {
+      return;
+    }
+    if (!req.user) {
+      throw fastify.httpErrors.unauthorized();
+    }
+  });
+
+  fastify.get('/api/v1/auth/user', async (req, reply) => {
+    if (req.user) {
+      reply.header('Content-Type', 'application/json').send(req.user);
```

Comment:
[nit] Would just returning `req.user` as an object work here?
```suggestion
      return req.user;
```